store/src/java/com/zimbra/cs/service/PreAuthServlet.java

```diff
@@ -330,12 +333,39 @@ private void redirectToCorrectServer(HttpServletRequest req, HttpServletResponse
     private static final String DEFAULT_MAIL_URL = "/zimbra";
     private static final String DEFAULT_ADMIN_URL = "/zimbraAdmin";
 
+    private static boolean isPrivate172SeriesAddress(String domain) {
+        int address = InetAddresses.coerceToInteger(InetAddresses.forString(domain));
+        return (((address >>> 24) & 0xFF) == 10)
+                || ((((address >>> 24) & 0xFF) == 172)
+                  && ((address >>> 16) & 0xFF) >= 16
+                  && ((address >>> 16) & 0xFF) <= 31)
+                || ((((address >>> 24) & 0xFF) == 192)
+                  && (((address >>> 16) & 0xFF) == 168));
+    }
+
     private void setCookieAndRedirect(HttpServletRequest req, HttpServletResponse resp, AuthToken authToken) throws IOException, ServiceException {
         boolean isAdmin = AuthToken.isAnyAdmin(authToken);
         boolean secureCookie = req.getScheme().equals("https");
         authToken.encode(resp, isAdmin, secureCookie);
 
         String redirectURL = getOptionalParam(req, PARAM_REDIRECT_URL, null);
+        URI uri = null;
+        try {
+            uri = new URI(redirectURL);
+        } catch (URISyntaxException exp) {
+            ZimbraLog.account.debug(String.format("URI %s is a malformed URL", redirectURL), exp);
+        }
+        String domain = uri.getHost();
```

Comment:
This line can result in NullPointerException following to eating the exception above.